src/alloc.rs

```diff
@@ -109,6 +110,53 @@ unsafe impl GlobalAlloc for GCMalloc {
     }
 }
 
+pub struct GCMalloc;
+
+unsafe impl Alloc for GCMalloc {
+    unsafe fn alloc(&mut self, layout: Layout) -> Result<NonNull<u8>, AllocErr> {
+        let ptr = libc::malloc(layout.size() as libc::size_t) as *mut u8;
+
+        ALLOC_LOCK.lock();
+        match ALLOC_INFO {
+            Some(ref mut pm) => pm.insert(ptr as usize, layout.size(), true),
+            None => {
+                let mut al = AllocList::new();
+                al.insert(ptr as usize, layout.size(), true);
+                ALLOC_INFO = Some(al);
+            }
+        };
+        ALLOC_LOCK.unlock();
+        Ok(NonNull::new_unchecked(ptr))
+    }
+
+    unsafe fn dealloc(&mut self, ptr: NonNull<u8>, _layout: Layout) {
+        libc::free(ptr.as_ptr() as *mut libc::c_void);
+        ALLOC_LOCK.lock();
+        ALLOC_INFO.as_mut().unwrap().remove(ptr.as_ptr() as usize);
+        ALLOC_LOCK.unlock();
+    }
+
+    unsafe fn realloc(
+        &mut self,
+        ptr: NonNull<u8>,
+        _layout: Layout,
+        new_size: usize,
+    ) -> Result<NonNull<u8>, AllocErr> {
+        let new_ptr = libc::realloc(ptr.as_ptr() as *mut libc::c_void, new_size) as *mut u8;
+
+        ALLOC_LOCK.lock();
+        // In all other cases, the memory block is resized and a new ptr
```

Comment:
I'm not quite sure how to parse "in all other cases" -- it feels like there's a missing sentence preceding this?

Reply:
This was copied over in error. I've removed it.